editor/code/src/goals.ts

```diff
@@ -86,10 +86,11 @@ export class InfoPanel {
   // notify the info panel that we have fresh goals to render
   requestDisplay(goals: GoalAnswer<PpString>) {
     this.postMessage("renderGoals", goals);
-    let vizx = extensions.getExtension("inQWIRE.vizx");
-    if (vizx?.isActive) {
-      commands.executeCommand("vizx.lspRender", goals);
-    }
+  }
+
+  requestVizxDisplay(goals: GoalAnswer<PpString>) {
```

Comment:
I believe that doesn't actually typecheck because the `GoalAnswer` returned by sendRequest is parameterized by `PpString` and so it needs the more general type.

Reply:
Yes, to properly type this would require a bit of hackery with phamtom types as done in the LSP protocol  typescript itself, we can give a try at some point.